cmake/XrplSanitizers.cmake

```diff
@@ -0,0 +1,174 @@
+separate_arguments(_san_list)
+
+set(REMAINING_SANITIZERS "")
+foreach(_san IN LISTS _san_list)
+  if(_san STREQUAL "Address")
+    set(ENABLE_ASAN TRUE)
+  elseif(_san STREQUAL "Thread")
+    set(ENABLE_TSAN TRUE)
+  elseif(_san STREQUAL "UndefinedBehavior")
+    set(ENABLE_UBSAN TRUE)
+  else()
+    list(APPEND REMAINING_SANITIZERS "${_san}")
+  endif()
+endforeach()
+
+if(REMAINING_SANITIZERS)
+  message(FATAL_ERROR
+    "Unknown SANITIZERS value(s): ${REMAINING_SANITIZERS}. "
+    "Supported: Address, Thread, UndefinedBehavior and their combinations.")
+endif()
+
+# Detect compiler type
+set(IS_GCC FALSE)
+set(IS_CLANG FALSE)
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    set(IS_GCC TRUE)
+    message(STATUS "  Compiler: GCC ${CMAKE_CXX_COMPILER_VERSION}")
+elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    set(IS_CLANG TRUE)
+    message(STATUS "  Compiler: Clang ${CMAKE_CXX_COMPILER_VERSION}")
+endif()
+
+# Detect platform (amd64/x86_64 vs arm64/aarch64)
+set(IS_AMD64 FALSE)
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64|AMD64")
+    set(IS_AMD64 TRUE)
+    message(STATUS "  Platform: amd64")
+else()
+    message(STATUS "  Platform: ${CMAKE_SYSTEM_PROCESSOR}")
+endif()
+
+# Frame pointer is required for meaningful stack traces. Sanitizers recommend minimum of -O1 for reasonable performance
+set(SANITIZERS_COMPILE_FLAGS "-fno-omit-frame-pointer" "-O1")
+
+# Build the sanitizer flags list
+set(SANITIZERS_FLAGS)
+
+if(ENABLE_ASAN)
+    list(APPEND SANITIZERS_FLAGS "address")
+elseif(ENABLE_TSAN)
+    list(APPEND SANITIZERS_FLAGS "thread")
+endif()
+
+if(ENABLE_UBSAN)
+    # UB sanitizer flags
+    if(IS_CLANG)
+        # Clang supports additional UB checks
+        list(APPEND SANITIZERS_FLAGS "undefined" "float-divide-by-zero" "unsigned-integer-overflow")
+    else()
+        list(APPEND SANITIZERS_FLAGS "undefined" "float-divide-by-zero")
+    endif()
+endif()
+
+# Configure code model for GCC on amd64
+# Use large code model for ASAN to avoid relocation errors
+# Use medium code model for TSAN (large is not compatible with TSAN)
+set(SANITIZERS_RELOCATION_FLAGS "")
+if(IS_GCC AND IS_AMD64)
+    if(ENABLE_ASAN)
+        message(STATUS "  Using large code model (-mcmodel=large)")
+        list(APPEND SANITIZERS_COMPILE_FLAGS "-mcmodel=large")
+        set(SANITIZERS_RELOCATION_FLAGS "-mcmodel=large")
+    elseif(ENABLE_TSAN)
+        message(STATUS "  Using medium code model (-mcmodel=medium)")
+        list(APPEND SANITIZERS_COMPILE_FLAGS "-mcmodel=medium")
+        set(SANITIZERS_RELOCATION_FLAGS "-mcmodel=medium")
+    endif()
+endif()
+
+# Compiler-specific configuration
+if(IS_GCC)
+    # Remove any -fuse-ld=... flags previously added to 'common'
+    get_target_property(_common_if_libs common INTERFACE_LINK_LIBRARIES)
+    if(_common_if_libs)
+        list(REMOVE_ITEM _common_if_libs "-fuse-ld=mold" "-fuse-ld=gold" "-fuse-ld=lld")
+        set_target_properties(common PROPERTIES INTERFACE_LINK_LIBRARIES "${_common_if_libs}")
+    endif()
```

Comment:
I think it would be better to not set these in the first place, if `SANITIZERS` is not empty